File: leetcode/go/3217.delete-nodes-from-linked-list-present-in-array.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func modifiedList(nums []int, head *ListNode) *ListNode {
    dummy := ListNode {
        Val: 0,
        Next: head,
    }

    set := make(map[int]bool)
    for _, n := range nums {
        set[n] = true
    }

    curr := &dummy

    for curr.Next != nil {
        if _, ok := set[curr.Next.Val]; ok {
            curr.Next = curr.Next.Next
        } else {
            curr = curr.Next
        }
    }

    return dummy.Next
}

File: leetcode/python/3217.delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0, head)

        numSet = set(nums)

        curr = dummy

        while curr.next is not None:
            if curr.next.val in numSet:
                curr.next = curr.next.next
            else:
                curr = curr.next
        
        return dummy.next

